processing/create_mosaic.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+#
+# XX
+#
+# USAGE:
+#   ${SCT_DIR}/python/bin/python XX
+#
+#   Example:
+#     XX
+#
+# DEPENDENCIES:
+#   SCT
+#
+
+import os
+import glob
+import argparse
+import sys
+import numpy as np
+import nibabel as nib
+import matplotlib.pyplot as plt
+from skimage.exposure import equalize_adapthist
+
+from spinalcordtoolbox.image import Image
+import spinalcordtoolbox.reports.slice as qcslice
+import sct_utils as sct
+
+
+def equalized(a):
+    """
+    Perform histogram equalization using CLAHE.
+
+    Note: function copied/pasted from spinalcordtoolbox.reports.qc
+    """
+    winsize = 16
+    min_, max_ = a.min(), a.max()
+    b = (np.float32(a) - min_) / (max_ - min_)
+    b[b >= 1] = 1  # 1+eps numerical error may happen (#1691)
+
+    h, w = b.shape
+    h1 = (h + (winsize - 1)) // winsize * winsize
+    w1 = (w + (winsize - 1)) // winsize * winsize
+    if h != h1 or w != w1:
+        b1 = np.zeros((h1, w1), dtype=b.dtype)
+        b1[:h, :w] = b
+        b = b1
+    c = equalize_adapthist(b, kernel_size=(winsize, winsize))
+    if h != h1 or w != w1:
+        c = c[:h, :w]
+    return np.array(c * (max_ - min_) + min_, dtype=a.dtype)
+
+
+def main():
+    # find all the images of insterest and store the mid slice in slice_lst
+    slice_lst = []
+    for x in os.walk(i_folder):
+        for file in glob.glob(os.path.join(x[0],"*"+im_suffixe)):
+
+            # load data
+            if plane == 'ax':
+                file_seg = file.split('.nii.gz')[0]+'_'+seg_suffixe
+
+                qcslice_cur = qcslice.Axial([Image(file), Image(file_seg)])
+                center_x_lst, center_y_lst = qcslice_cur.get_center()  # find seg center of mass
+                mid_slice_idx = int(qcslice_cur.get_dim(qcslice_cur._images[0]) // 2)  # find index of the mid slice
+                mid_slice = qcslice_cur.get_slice(qcslice_cur._images[0].data, mid_slice_idx)  # get the mid slice
+                # crop image around SC seg
+                mid_slice = qcslice_cur.crop(mid_slice,
+                                            int(center_x_lst[mid_slice_idx]), int(center_y_lst[mid_slice_idx]),
+                                            30, 30)
+            else:
+                qcslice_cur = qcslice.Sagittal([Image(file)])
+                mid_slice_idx = int(qcslice_cur.get_dim(qcslice_cur._images[0]) // 2)  # find index of the mid slice
+                mid_slice = qcslice_cur.get_slice(qcslice_cur._images[0].data, mid_slice_idx)  # get the mid slice
```

Comment:
@Nicolas-Pinon : For some reason (that I can't find), `mid_slice` is empty while `Image(file)` is not.
This is in the sagittal scenario, with this kind of command:
```
python create_mosaic.py -i T1w_flatten.nii.gz -ifolder ~/amu_spineGeneric/ -p sag -n 3 -o mosaic_sag_3_2.png
```

Note --> I have run `sct_flatten_sagittal` on a BIDS dataset before using it as testing data.

Reply:
> For some reason (that I can't find), mid_slice is empty while Image(file) is not.

Maybe it's the same old bug where opening nii instead of nii.gz creates empty data?

Reply:
> Note --> I have run sct_flatten_sagittal on a BIDS dataset before using it as testing data.

I am currently copying the latest processed data. It will be available in ~30min at:
```duke:/projects/spine_generic/spineGeneric_20190513/data/```

Reply:
Ok! Will work with the last dataset!